src/cr/cube/cube_slice.py

```diff
@@ -26,134 +54,88 @@ def _update_args(self, kwargs):
             return kwargs
 
         axis = kwargs.get('axis')
-        if axis is None or isinstance(axis, tuple):
-            # If no axis was passed, we don't need to update anything. If axis
-            # was a tuple, it means that the caller was expecting a 3D cube.
-            return kwargs
+        if isinstance(axis, int):
+            kwargs['axis'] += 1
+
+        hs_dims_key = (
+            'transforms'
+            if 'transforms' in kwargs else
+            'include_transforms_for_dims'
+        )
+        hs_dims = kwargs.get(hs_dims_key)
+        if isinstance(hs_dims, list):
+            kwargs[hs_dims_key] = [dim + 1 for dim in hs_dims]
 
-        kwargs['axis'] += 1
         return kwargs
 
     def _update_result(self, result):
         if self.ndim < 3 or len(result) - 1 < self._index:
             return result
         result = result[self._index]
+        if isinstance(result, tuple):
+            return list(result)
```

Comment:
Why not convert to `np.array`? and make the `if` below an `elif`?

Reply:
Done